File: terra.py
import math

def calcular_curvatura(distancia_km):
    """
    Calcula a curvatura da Terra (queda vertical) para uma determinada distância.

    Args:
        distancia_km (float): A distância em quilômetros.

    Returns:
        float: A queda vertical em metros.
    """
    # Raio médio da Terra em quilômetros
    raio_terra_km = 6371.0

    # Calcula a queda vertical em quilômetros
    queda_vertical_km = (distancia_km**2) / (2 * raio_terra_km)

    # Converte a queda vertical para metros para uma leitura mais fácil
    queda_vertical_metros = queda_vertical_km * 1000

    return queda_vertical_metros

def main():
    """
    Função principal para executar o programa.
    """
    print("--- Calculadora de Curvatura da Terra ---")
    print("Este programa calcula a queda vertical (curvatura) da Terra ")
    print("para uma determinada distância.")
    print("------------------------------------------")

    try:
        distancia = float(input("Digite a distância em quilômetros: "))
        if distancia < 0:
            print("Por favor, digite um valor positivo.")
            return

        curvatura_em_metros = calcular_curvatura(distancia)

        print(f"\nPara uma distância de {distancia} km, a curvatura (queda vertical) é de:")
        print(f"{curvatura_em_metros:.2f} metros.")
        print("\n*Este cálculo é uma aproximação e desconsidera fatores como refração atmosférica.")

    except ValueError:
        print("Entrada inválida. Por favor, digite um número.")

if __name__ == "__main__":
    main()

File: terra.c
#include <stdio.h>
#include <math.h>

// Define a constante para o raio médio da Terra em quilômetros.
#define RAIO_TERRA_KM 6371.0

// Protótipo da função.
double calcular_curvatura(double distancia_km);

int main() {
    double distancia_km;
    double curvatura_metros;

    printf("--- Calculadora de Curvatura da Terra ---\n");
    printf("Este programa calcula a queda vertical (curvatura) da Terra \n");
    printf("para uma determinada distancia.\n");
    printf("------------------------------------------\n");

    printf("Digite a distancia em quilometros: ");

    // A função scanf retorna o número de itens lidos com sucesso.
    // Usar essa verificação evita comportamentos inesperados em caso de entrada inválida.
    if (scanf("%lf", &distancia_km) != 1 || distancia_km < 0) {
        printf("Entrada invalida. Por favor, digite um numero positivo.\n");
        return 1; // Retorna um código de erro.
    }

    curvatura_metros = calcular_curvatura(distancia_km);

    printf("\nPara uma distancia de %.2lf km, a curvatura (queda vertical) e de:\n", distancia_km);
    printf("%.2f metros.\n", curvatura_metros);
    printf("\n*Este calculo e uma aproximacao e desconsidera fatores como refracao atmosferica.\n");

    return 0; // Indica que o programa terminou com sucesso.
}

/**
 * @brief Calcula a curvatura da Terra (queda vertical) para uma determinada distancia.
 * @param distancia_km A distancia em quilometros.
 * @return A queda vertical em metros.
 */
double calcular_curvatura(double distancia_km) {
    // Calcula a queda vertical em quilômetros.
    double queda_vertical_km = (distancia_km * distancia_km) / (2 * RAIO_TERRA_KM);

    // Converte a queda vertical para metros.
    double queda_vertical_metros = queda_vertical_km * 1000;

    return queda_vertical_metros;
}